### DISCUSSION TITLE
Can it be used with load balancing?

### DISCUSSION THREAD

**Discussion by daecide** [nextauthjs/next-auth#1249] [Help] [answered]
Can i use next-auth for my frontend If i have load balancing on it?
I mean, i am creating a website using next for frontend.
For security purpose, next-auth have this cool secure built-in session and i wanted to use it for my authentication.
Maybe someone else have ask of this issue or maybe it's an obvious answer, but I've been looking through the documentation and googling for this issue but i haven't found it.
If it can be used with load balancing, do i have to make adjustment? if so, how?

Thanks guys!

**Comment by StefanSelfTaught** [comment, marked as answer]
You can use a load balancer. However if you are making use of some deployment platform who supports serverless you might not need it. The files in your `api` folder can be deployed as serverless functions.

**Reply by iaincollins** [reply]
Hi there! Happy to confirm this is a good answer.

NextAuth.js avoids problems like using file system or host specific configuration see in some older session management libraries (e.g. the sort of behaviour you might associate with using sessions in PHP), so is scalable behind a load balancer without any additional configuration.

You'll want to make sure you allow your database allows enough connection to handle the multiple requests (also an issue with Serverless functions), at least 2 x the number of servers (or instances of Serverless functions) you have is probably sensible to avoid running into problems during deployments / updates (when the number of connections may spike).

If you are using JSON Web Tokens you can drastically reduce the amount of open connections needed, as it will only need to contact the database during sign in and not for session validation (using JSON Web Tokens likely result in a faster user experience too).

**Reply by geminiyellow** [reply]
can i get same sns logged in status with load balancing?
i only use building provider but not adapter in it.
